doc/http-api.md

```diff
@@ -0,0 +1,101 @@
+  If all goes well, the enclave responds with status code `200`.
+
+* `GET /enclave/config` Returns nitriding's configuration.  
+  The enclave responds with status code `200`.
+
+## External endpoints, reachable to other enclaves
+
+* `GET /enclave/sync?nonce={nonce}` Exposed by workers, the leader talks to this endpoint to initiate key synchronization.  
+  `nonce` must be a 20-byte nonce encoded in 40 hexadecimal digits.
+  If all goes well, the worker responds with status code `200` and the following JSON-formatted body:
+  ```
+  {
+    "document": "{Base64-encoded attestation document}",
+  }
+  ```
+
+* `POST /enclave/sync` Exposed by workers, the leader talks to this endpoint to
+  complete key synchronization.  
+
+  The request must contain the following JSON-formatted body:
+  ```
+  {
+    "document": "{Base64-encoded attestation document}",
+    "encrypted_keys": "{Base64-encoded, encrypted enclave keys}",
+  }
+  ```
+  If all goes well, the worker responds with status code `200`.
+
+* `POST /enclave/heartbeat` Exposed by the leader, workers periodically send a heartbeat to this endpoint.  
+  The request must contain the following JSON-formatted body:
+  ```
+  {
+    "hashed_keys": "{hashed_keys}",
+    "worker_hostname": "{worker_hostname}",
+  }
+  ```
+  `worker_hostname` contains the worker's EC2-internal hostname, e.g., `ip-12-34-56-78.us-east-2.compute.internal`.  
+  `hashed_keys` contains the Base64-encoded SHA-256 hash over the worker's enclave key material.
+  If all goes well, the leader responds with status code `200`.
+
+* `GET /enclave/leader?nonce={nonce}` Exposed by all enclaves, this endpoint
+  helps enclaves figure out who the leader is.  
+  `nonce` must be a 20-byte nonce encoded in 40 hexadecimal digits.
+  All enclaves create a random `nonce` and send it to the leader's endpoint.
+  If the leader notices that it's talking to itself (by comparing the received nonce to its previously-generated nonce),
+  it designated itself as the leader.
+  After that, the leader responds with status code `410`.
+  Before that, the leader responds with status code `200`.
+  While workers expose this endpoint too, they should never receive any requests.
+
+## Internal endpoints, reachable to the application
+
+* `GET /enclave/ready` Invoked by the application to signal its readiness.  
+  When nitriding is invoked with the command line argument `-wait-for-app`,
+  it refrains from starting its external Web servers until the application
+  signals its readiness by calling this endpoint, after which nitriding starts
+  the external Web servers.
+
+* `GET /enclave/state` Returns the application's state in the response body.  
+  This endpoint allows an application to retrieve state
+  (e.g., confidential key material) that was previously set by the "leader" application.
+  If synchronization is not enabled via the `-fqdn-leader` command line
+  argument, the endpoint responds with status code `410`.
```

Comment:
Maybe return `403 Forbidden` here to distinguish from the "I'm the leader" case? The resource hasn't been removed; it was never enabled in the first place.